cql3/untyped_result_set.hh

```diff
@@ -67,8 +77,8 @@ public:
         return *_data.at(name);
     }
     template<typename T>
-    T get_as(const sstring& name) const {
-        return value_cast<T>(data_type_for<T>()->deserialize(get_blob(name)));
+    typename cast_type_for<T>::type get_as(const sstring& name) const {
+        return value_cast<typename cast_type_for<T>::type>(data_type_for<T>()->deserialize(get_blob(name)));
```

Comment:
I don't understand this change. What are you trying to achieve with this change?

Reply:
It is a temporary change to allow get_as<date_type_native_type>. It is removed in a followup patch.

The idea is to first require all code to use date_type_native_type or timestamp_type_native_type. That way we know we are not accidentally changing code that really wants to use date_type.